File: C++/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <stack>
#include <iostream>

using namespace std;

bool solution(string s)
{
    stack<char> myStack;
    bool answer = true;
    
    for (auto it=s.begin(); it!=s.end(); it++)
    {
        if (*it == '(')
        {
            myStack.push(*it);
        }
        else
        {
            if (!myStack.empty() && myStack.top() == '(')
            {
                myStack.pop();
            }
            else
            {
                answer = false;
                return answer;
            }
        }
    }

    if (!myStack.empty()) {answer = false;}
    return answer;
}

File: Python3/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    answer = True
    q = deque()
    
    for ch in s :
        if ch == '(' : q.append('(')
        else :
            if len(q) != 0 :
                q.pop()
            else :
                answer = False
                return answer
        
    if len(q) == 0 : answer = True
    else : answer = False

    return answer